pageserver/src/import_datadir.rs

```diff
@@ -705,8 +642,9 @@ pub async fn create_tar_zst(pgdata_path: &Utf8Path) -> Result<Vec<u8>> {
     }
     // Do a sort to get a more consistent listing
     paths.sort_unstable();
+    let tempfile = File::from_std(tempfile()?);
```

Comment:
This creates a file in the default tempfile location, i.e., `/tmp`.tmpfs.
We don't want that. We want all files to sit inside the tenant that they belong to, on the `/storage` partition.
I suggest using the `TEMP_FILE_SUFFIX` like we do for other temp files.
You'll need to do the cleanup manually as follows:
1. the load procedure should to the cleanup (`fn reconcile()` is a good starting point to learn more)
2. unlink an existing file here before creating it, and create with `create_new` to detect race conditions

I'm not a fan of `unlink()` directly `create_new()` , like `tempfile()` does.
The reason is that the inode will still grow on disk but we don't see that usage using user-space tools like `du`.

I'd also like this to be a VirtualFile, but, we don't have a tempfile abstraction for VirtualFile yet afaik. The primary challenge there is that the `create` + unlink

Reply:
There is no `AsyncRead`/`AsyncWrite` impls for VirtualFile so one would have to re-implement all the copy functions. I understand the need to be inside the fd limitation though.

Reply:
You're misreading my comment. VirtualFile is nice to have I think.
The `tempfile()` thing is the blocker here.

Reply:
camino_tempfile does also have a builder, but we can go with a static tempfile here as we are finally guarded by a mutex! :)

Reply:
I have changed it to use a file from the timelines dir, so I'm resolving this.